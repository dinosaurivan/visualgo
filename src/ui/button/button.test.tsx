// libraries 
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";

// components
import { Button } from "./button";



describe(
  "Button UI component tests", () => {
    
    test(
      "renders correctly with text provided", () => {
        
        render(<Button text="Click me" />);
        const component = screen.getByRole("button");
        
        expect(component).toBeInTheDocument();
        expect(component).toMatchSnapshot();
        
        expect(component).not.toBeDisabled();
        expect(component).toHaveTextContent("Click me");
        expect(component).toHaveAttribute("type", "button");
        expect(component).toHaveClass("text text_type_button text_color_primary button");
      }
    );
    
    test(
      "renders correctly without text provided", () => {
        
        render(<Button />);
        const component = screen.getByRole("button");
        
        expect(component).toBeInTheDocument();
        expect(component).toMatchSnapshot();
        
        expect(component).not.toBeDisabled();
        expect(component).toHaveTextContent("");
        expect(component).toHaveAttribute("type", "button");
        expect(component).toHaveClass("text text_type_button text_color_primary button");
      }
    );    
    
    test(
      "calls onClick handler when is clicked on", async () => {
        
        const user = userEvent.setup()
        const onClickMock = jest.fn();
        render(<Button text="Click me" onClick={onClickMock} />);
        const component = screen.getByRole("button");
        
        await user.click(component);
        
        expect(component).toBeInTheDocument();
        expect(component).toMatchSnapshot();
        
        expect(onClickMock).toHaveBeenCalledTimes(1);
      }
    );
    
    test(
      "disables when disabled prop is set to true", () => {
        
        render(<Button text="Click me" disabled />);
        const component = screen.getByRole("button");
        
        expect(component).toBeInTheDocument();
        expect(component).toMatchSnapshot();
        
        expect(component).toBeDisabled();
      }
    );
    
    test(
      "disables when isLoader prop is set to true", () => {
        
        render(<Button text="Click me" isLoader={true} />);
        const component = screen.getByRole("button");
        
        expect(component).toBeInTheDocument();
        expect(component).toMatchSnapshot();
        
        expect(component).toBeDisabled();
        expect(component).toHaveClass("loader");
      }
    );    
  }
);
