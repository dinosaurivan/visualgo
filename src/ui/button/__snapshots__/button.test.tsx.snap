// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI component tests calls onClick handler when is clicked on 1`] = `
<button
  class="text text_type_button text_color_primary button   "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button UI component tests disables when disabled prop is set to true 1`] = `
<button
  class="text text_type_button text_color_primary button   "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button UI component tests disables when isLoader prop is set to true 1`] = `
<button
  class="text text_type_button text_color_primary button  loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button UI component tests renders correctly with text provided 1`] = `
<button
  class="text text_type_button text_color_primary button   "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button UI component tests renders correctly without text provided 1`] = `
<button
  class="text text_type_button text_color_primary button   "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
